.adr/0009-nitro-service.md

```diff
@@ -0,0 +1,112 @@
+---
+title: nitro-api
+tags: nitro, api, rpc
+author: Louis
+source: 
+---
+
+# 0009 - Nitro as a service API definition
+
+## Status
+
+proposed
+
+## Context
+
+Objectives are a powerful abstraction for multi-party protocols, but when using nitro as a library it can be time consuming and cumbersome for new app developers.
+
+Virtual application using voucher abstractions are the most lightweight and autonomous way of interacting with Nitro channels.
+
+Nitro should abstract the virtual channel plumbing to the app developer and provide a simple interface to provision various channel construction protocols.
+
+## Decision
+
+Define the API features to be exposed using various encoders and transports such as:
+- HTTP2
+- Websocket
+- NATS
+- TCP wire
+
+Encoding:
+- JSON/BSON
+- MsgPack
+- Protobuf
+
+### Notation
+
+We will use the following notation to describe the API, but they can be implemented in different ways.
+
+This notation remains the recommended serialization for JSON, BSON, MsgPack, Protobuf, it is more compact than json-rpc and compatible with Message Queue systems.
+
+#### Request:
+`[type, msgid, method, params]`
+
+#### Response:
+`[type, msgid, error, result]`
+
+#### Event:
+`[type, msgid, method, params]`
+
+##### type
+The message type, must be the integer zero (0) for "Request" messages. 
+One (1) means that this message is the "Response" message.
+Two (2) means that this message is the "Notification" message.
```

Comment:
Is this similar to the JSON-RPC [notifications](https://www.jsonrpc.org/specification#notification) where "notifications" is just a request from a client where the client doesn't care about the response?